File: backend-node/src/admin/admin.controller.ts
import { Controller, Get, Query } from '@nestjs/common';
import { Roles } from '../auth/http-auth.decorators';
import { JavaClientService } from '../shared/java-client.service';
import { ListAuditLogDto } from './dto/list-audit-log.dto';

@Controller('admin')
@Roles('ADMIN')
export class AdminController {
  constructor(private readonly javaClient: JavaClientService) {}

  @Get('overview')
  overview(): Promise<unknown> {
    return this.javaClient.get('/admin/overview');
  }

  @Get('audit-logs')
  auditLogs(@Query() query: ListAuditLogDto): Promise<unknown> {
    const parameters = new URLSearchParams({
      page: String(query.page),
      size: String(query.size),
    });
    if (query.action) parameters.set('action', query.action);
    if (query.outcome) parameters.set('outcome', query.outcome);
    if (query.username) parameters.set('username', query.username);

    return this.javaClient.get(
      `/admin/audit-logs?${parameters.toString()}`,
    );
  }
}

File: backend-java/src/main/java/com/uav/backend/audit/AdminController.java
package com.uav.backend.audit;

import com.uav.backend.audit.dto.AdminOverviewResponse;
import com.uav.backend.audit.dto.AuditLogPageResponse;
import com.uav.backend.audit.service.AuditLogService;
import com.uav.backend.common.ApiResponse;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/admin")
public class AdminController {

    private final AuditLogService auditLogService;

    public AdminController(AuditLogService auditLogService) {
        this.auditLogService = auditLogService;
    }

    @GetMapping("/overview")
    public ApiResponse<AdminOverviewResponse> overview() {
        return ApiResponse.ok(auditLogService.overview());
    }

    @GetMapping("/audit-logs")
    public ApiResponse<AuditLogPageResponse> auditLogs(
            @RequestParam(defaultValue = "0") int page,
            @RequestParam(defaultValue = "20") int size,
            @RequestParam(required = false) String action,
            @RequestParam(required = false) String outcome,
            @RequestParam(required = false) String username) {
        return ApiResponse.ok(auditLogService.search(
                page,
                size,
                action,
                outcome,
                username
        ));
    }
}
